### PR TITLE
HHH-10358

### PR DESCRIPTION
**Pull request by stalep**
try to avoid calling the syncronized method

### CODE REVIEW

**Comment by stalep** [comment]
so i created two pr's one for master and another for 5.0, ignore this if thats the wrong way of doing it :)

File: hibernate-core/src/main/java/org/hibernate/loader/plan/exec/internal/AbstractLoadPlanBasedLoader.java [hunk 1]
```diff
@@ -431,7 +431,8 @@ protected final ResultSet getResultSet(
 
 		try {
 			ResultSet rs = session.getJdbcCoordinator().getResultSetReturn().extract( st );
-			rs = wrapResultSetIfEnabled( rs , session );
+			if(session.getFactory().getSessionFactoryOptions().isWrapResultSetsEnabled())
```
Comment (Sanne) [inline hibernate-core/src/main/java/org/hibernate/loader/plan/exec/internal/AbstractLoadPlanBasedLoader.java]: Unless this is something which can be changed at runtime, shouldn't we cache this as a final boolean field of this instance?

BTW your style is still non-hibernate; see https://github.com/hibernate/hibernate-ide-codestyles/

File: hibernate-core/src/main/java/org/hibernate/loader/plan/exec/internal/AbstractLoadPlanBasedLoader.java [hunk 2]
```diff
@@ -471,22 +472,17 @@ protected void advance(final ResultSet rs, final RowSelection selection) throws
...
-						.getResultSetWrapper().wrap( rs, retreiveColumnNameToIndexCache( rs ) );
-			}
-			catch(SQLException e) {
-				log.unableToWrapResultSet( e );
-				return rs;
+		try {
+			if ( log.isDebugEnabled() ) {
+				log.debugf( "Wrapping result set [%s]", rs );
 			}
+			return session.getFactory()
```
Comment (Sanne) [inline hibernate-core/src/main/java/org/hibernate/loader/plan/exec/internal/AbstractLoadPlanBasedLoader.java]: A great deal of these operations could be moved out of the synchronized block, even when wrapping is enabled.

I'll take your idea and go one step further, I hope you don't mind ;-)

**Comment by Sanne** [comment]
Thanks, I'll close this and massage #1177 then apply it to both branches.

**Pull request closed without merging**